File: baekjoon/boj_1260.py
from collections import deque

n, m, v = map(int, input().split())

graph = dict()

for i in range(1, n+1):
  graph[i] = list()

for i in range(1, m+1):
    start_v, end_v = map(int, input().split())
    graph[start_v].append(end_v)
    if start_v not in graph[end_v]:
        graph[end_v].append(start_v)    

for i in graph:
   graph[i].sort()

dfs_visited = list()
bfs_visited = list()
    
def dfs(start_v):
    dfs_visited.append(start_v)
    for next_v in graph[start_v]:
       if next_v not in dfs_visited:
          dfs(next_v)
    
def bfs(start_v):
    q = deque()
    bfs_visited.append(start_v)
    q.append(start_v)
    while q:
        cur_v = q.popleft()
        for next_v in graph[cur_v]:
          if next_v not in bfs_visited:
            q.append(next_v)
            bfs_visited.append(next_v)
dfs(v)
bfs(v)

print(*dfs_visited)
print(*bfs_visited)

File: src/baekjoon/Boj_1260.java
package baekjoon;

import java.util.*;
import java.io.*;
import java.util.stream.*;

class Boj_1260 {

    private static Map<Integer, List<Integer>> map = new HashMap<>();
    private static List<Integer> dfsVisited = new ArrayList<>();
    private static List<Integer> bfsVisited = new ArrayList<>();
    private static int n;
    private static int m;
    private static int v;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        for (int i = 1; i <= n; i++) {
            map.put(i, new ArrayList<>());
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int startV = Integer.parseInt(st.nextToken());
            int endV = Integer.parseInt(st.nextToken());
            addVertexAndSort(startV, endV);
            addVertexAndSort(endV, startV);
        }
        dfsVisited.add(v);
        dfs(v);
        bfs(v);

        System.out.println(
                dfsVisited.stream()
                        .map(String::valueOf)
                        .collect(Collectors.joining(" "))
        );

        System.out.println(
                bfsVisited.stream()
                        .map(String::valueOf)
                        .collect(Collectors.joining(" "))
        );
    }

    private static void addVertexAndSort(int startV, int endV) {
        List<Integer> vertexs = map.get(startV);
        vertexs.add(endV);
        Collections.sort(vertexs);
    }

    private static void dfs(int currentV) {
        for (Integer vertex : map.get(currentV)) {
            if (!dfsVisited.contains(vertex)) {
                dfsVisited.add(vertex);
                dfs(vertex);
            }
        }
    }

    private static void bfs(int startV) {
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(startV);
        bfsVisited.add(startV);
        while (!queue.isEmpty()) {
            int curV = queue.poll();
            for (Integer nextV : map.get(curV)) {
                if (!bfsVisited.contains(nextV)) {
                    queue.offer(nextV);
                    bfsVisited.add(nextV);
                }
            }
        }
    }
}
